driver/parallel.go

```diff
@@ -187,6 +187,11 @@ func (pg *parallelGroup) sourceToRequest(src Source) (*api.WorkerRequest, error)
 			return nil, err
 		}
 		req.Proc = b
+	} else {
+		// If there's no filter, then send a match-all proc since the
+		// client can't handle a null req.Proc.  XXX It seems like a
+		// null proc should mean return everything in the span.
+		req.Proc = json.RawMessage(`{"op":"FilterProc","filter":{"op":"MatchAll"}}`)
```

Comment:
How about allowing ast.SearchRequest.Proc to be null or unset?
```diff
diff --git a/api/api.go b/api/api.go
index 458f88c5..74c06cd6 100644
--- a/api/api.go
+++ b/api/api.go
@@ -43,7 +43,7 @@ type TaskEnd struct {
 
 type SearchRequest struct {
 	Space SpaceID         `json:"space" validate:"required"`
-	Proc  json.RawMessage `json:"proc" validate:"required"`
+	Proc  json.RawMessage `json:"proc,omitempty"`
 	Span  nano.Span       `json:"span"`
 	Dir   int             `json:"dir" validate:"required"`
 }
diff --git a/ast/unpack.go b/ast/unpack.go
index 17197a5b..06198ed3 100644
--- a/ast/unpack.go
+++ b/ast/unpack.go
@@ -421,10 +421,16 @@ func UnpackMap(custom Unpacker, m interface{}) (Proc, error) {
 
 // UnpackJSON transforms a JSON representation of a proc into an ast.Proc.
 func UnpackJSON(custom Unpacker, buf []byte) (Proc, error) {
+	if len(buf) == 0 {
+		return nil, nil
+	}
 	obj, err := joe.Unmarshal(buf)
 	if err != nil {
 		return nil, err
 	}
+	if obj == nil {
+		return nil, nil
+	}
 	proc, err := unpackProc(custom, obj)
 	if err != nil {
 		return nil, err
diff --git a/driver/compile.go b/driver/compile.go
index 38e8eb12..a3381823 100644
--- a/driver/compile.go
+++ b/driver/compile.go
@@ -43,7 +43,12 @@ func zbufDirInt(reversed bool) int {
 	return 1
 }
 
+var passProc = &ast.PassProc{Node: ast.Node{"PassProc"}}
+
 func programPrep(program ast.Proc, sortKey field.Static, sortReversed bool) (ast.BooleanExpr, ast.Proc) {
+	if program == nil {
+		return nil, passProc
+	}
 	ReplaceGroupByProcDurationWithKey(program)
 	if sortKey != nil {
 		setGroupByProcInputSortDir(program, sortKey, zbufDirInt(sortReversed))
@@ -177,8 +182,6 @@ func ensureSequentialProc(p ast.Proc) *ast.SequentialProc {
 	}
 }
 
-var passProc = &ast.PassProc{Node: ast.Node{"PassProc"}}
-
 // liftFilter removes the filter at the head of the flowgraph AST, if
 // one is present, and returns its ast.BooleanExpr and the modified
 // flowgraph AST. If the flowgraph does not start with a filter, it
diff --git a/driver/parallel.go b/driver/parallel.go
index ad59a3db..01e7a7a8 100644
--- a/driver/parallel.go
+++ b/driver/parallel.go
@@ -188,11 +188,6 @@ func (pg *parallelGroup) sourceToRequest(src Source) (*api.WorkerRequest, error)
 			return nil, err
 		}
 		req.Proc = b
-	} else {
-		// If there's no filter, then send a match-all proc since the
-		// client can't handle a null req.Proc.  XXX It seems like a
-		// null proc should mean return everything in the span.
-		req.Proc = json.RawMessage(`{"op":"FilterProc","filter":{"op":"MatchAll"}}`)
 	}
 	req.Dir = pg.mcfg.Order.Int()
 	return &req, nil
```

Reply:
cool